File: output/qwen_2.5_72b_instruct/avatar/algo_based_translation/java/python/atcoder_ABC152_E.py

import math
import sys

def modDiv(a, b, mod):
    return (a % mod) * modInv(b, mod) % mod

def modInv(x, mod):
    return modPow(x, mod - 2, mod)

def modPow(b, e, mod):
    ans = 1
    x = b % mod
    y = e
    while y != 0:
        if y & 1 == 1:
            ans = (ans * x) % mod
        x = (x * x) % mod
        y >>= 1
    return ans

def main():
    import sys
    input = sys.stdin.read().split()
    N = int(input[0])
    A = list(map(int, input[1:]))
    lcm = 1
    for ai in A:
        a = ai
        gcd = math.gcd(lcm, a)
        lcm = (lcm // gcd) * a
    mod = 1000000007
    modLcm = lcm % mod
    ans = 0
    for ai in A:
        ans = (ans + modDiv(modLcm, ai, mod)) % mod
    print(ans)

if __name__ == "__main__":
    main()

File: output/qwen_2.5_72b_instruct/avatar/algo_based_translation/python/java/atcoder_ABC152_E.java

import java.util.Scanner;
import java.math.BigInteger;

public class atcoder_ABC152_E {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int MOD = 1000000007;
        int N = scanner.nextInt();
        int[] A = new int[N];
        for (int i = 0; i < N; i++) {
            A[i] = scanner.nextInt();
        }
        long _lcm = 1;
        for (int a : A) {
            _lcm = lcm(_lcm, a);
        }
        long sum = 0;
        for (int s : A) {
            sum = (sum + modInverse(s, MOD)) % MOD;
        }
        long ans = (_lcm * sum) % MOD;
        System.out.println(ans);
    }

    public static long gcd(long a, long b) {
        while (b != 0) {
            long temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }

    public static long lcm(long a, long b) {
        return a / gcd(a, b) * b;
    }

    public static long modInverse(long a, long m) {
        return new BigInteger(Long.toString(a)).modInverse(BigInteger.valueOf(m)).longValue();
    }
}

